index.js

```diff
@@ -217,23 +215,22 @@ function updateDesigns(designPaths, callback) {
         throw new VError(msg);
     }
     // everything looks good, we're ready to update any designs that need it
+    var updateDesignFns = [];
     designPaths.forEach(function (designPath) {
         var design = _.get(designs, designPath);
+        var parts = designPath.split('.');
         if (!design) {
             log.warn('skipping unknown design ' + designPath);
         } else {
-            var parts = designPath.split('.');
-            updateDesign(parts[0] + ':' + parts[1], parts[2], design, function (error, okMessage) {
-                if (error) {
-                    errors.push(error);
-                    log.error(error.message);
-                } else {
-                    log.info(okMessage);
-                }
-            });
+            updateDesignFns.push(updateDesign.bind(this, parts[0] + ':' + parts[1], parts[2], design));
         }
     });
-    return callback(errors.length ? errors : null);
+    if (updateDesignFns.length === 0) {
+        callback(null, 'no design paths match a design found in couchdb directory');
```

Comment:
since we're returning an array of messages in the parallel case, should we return an array here? (maybe just putting that message as the single entry in an array)